core/trino-main/src/main/java/io/trino/sql/planner/optimizations/StreamPropertyDerivations.java

```diff
@@ -342,6 +342,13 @@ public StreamProperties visitExchange(ExchangeNode node, List<StreamProperties>
                     if (node.getPartitioningScheme().getPartitioning().getHandle().equals(FIXED_ARBITRARY_DISTRIBUTION)) {
                         return new StreamProperties(FIXED, Optional.empty(), false);
                     }
+                    // if this is a transaction and there are no arguments we are safe to provide empty optional as there will be only one stream
+                    if (node.getPartitioningScheme().getPartitioning().getHandle().getTransactionHandle().isPresent()) {
+                        return new StreamProperties(FIXED, Optional.of(node.getPartitioningScheme().getPartitioning().getArguments().stream()
+                                .map(ArgumentBinding::getColumn)
+                                .collect(toImmutableList()))
+                                .filter(x -> !x.isEmpty()), false);
```

Comment:
The pattern `Optional.of( expr ).filter( condition )` is clever, but IMO doesn't make the code more readable.